### PR TITLE
Fixed a unused variable compilation error.

### PR DESCRIPTION
**Pull request by jhmueller-huawei**
Happens with clang and gcc (on at least linux).

### CODE REVIEW

File: Gem/Code/Source/RHI/BindlessPrototypeExampleComponent.cpp [hunk 1]
```diff
@@ -671,8 +671,11 @@ namespace AtomSampleViewer
...
         Matrix4x4 worldToClipMatrix = m_viewToClipMatrix * worldToViewMatrix;
-        bool set = m_floatBuffer->AllocateOrUpdateBuffer(m_worldToClipHandle, static_cast<void*>(&worldToClipMatrix), static_cast<uint32_t>(sizeof(Matrix4x4)));
-        
+        [[maybe_unused]] bool set
+            = m_floatBuffer->AllocateOrUpdateBuffer(m_worldToClipHandle,
+                                                    static_cast<void *>(&worldToClipMatrix),
+                                                    static_cast<uint32_t>(sizeof(Matrix4x4)));
+
         // Update the light direction
         set = m_floatBuffer->AllocateOrUpdateBuffer(m_lightDirectionHandle, static_cast<void*>(&m_lightDir), static_cast<uint32_t>(sizeof(Vector3)));
```
Comment (lumberyard-employee-dm) [inline Gem/Code/Source/RHI/BindlessPrototypeExampleComponent.cpp:680]: The `set` variable can just be removed
It is unused in both assignments.

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Gem/Code/Source/RHI/BindlessPrototypeExampleComponent.cpp (hunk 1 above)
Comment (VickyAtAZ) [reply Gem/Code/Source/RHI/BindlessPrototypeExampleComponent.cpp:680]: Agree

**Review by invertednormal — APPROVED** [review]
Approving but also agree with @lumberyard-employee-dm that the variable `set` should just be removed.

**Review by VickyAtAZ — COMMENTED** [review]
Prefer remove the 'set' variable instead of adding the maybe_unused.

File: Gem/Code/Source/RHI/BindlessPrototypeExampleComponent.cpp (hunk 1 above)
Comment (lumberyard-employee-dm) [inline Gem/Code/Source/RHI/BindlessPrototypeExampleComponent.cpp:680]:
```suggestion
        m_floatBuffer->AllocateOrUpdateBuffer(m_lightDirectionHandle, static_cast<void*>(&m_lightDir), static_cast<uint32_t>(sizeof(Vector3)));
```

File: Gem/Code/Source/RHI/BindlessPrototypeExampleComponent.cpp [hunk 2]
```diff
@@ -671,8 +671,11 @@ namespace AtomSampleViewer
 
         // Update the worldToClipMatrix 
         Matrix4x4 worldToClipMatrix = m_viewToClipMatrix * worldToViewMatrix;
-        bool set = m_floatBuffer->AllocateOrUpdateBuffer(m_worldToClipHandle, static_cast<void*>(&worldToClipMatrix), static_cast<uint32_t>(sizeof(Matrix4x4)));
-        
+        [[maybe_unused]] bool set
+            = m_floatBuffer->AllocateOrUpdateBuffer(m_worldToClipHandle,
```
Comment (lumberyard-employee-dm) [inline Gem/Code/Source/RHI/BindlessPrototypeExampleComponent.cpp:675]:
```suggestion
        m_floatBuffer->AllocateOrUpdateBuffer(m_worldToClipHandle,
```

**Pull request merged**